utils/docker/Dockerfile.centos.7

```
@@ -181,6 +173,13 @@ RUN mkdir -p /opt/daos /mnt/daos /var/run/daos_server /var/run/daos_agent &&   \
     chown daos_server.daos_server /opt/daos /mnt/daos /var/run/daos_server &&  \
     chown daos_agent.daos_agent /var/run/daos_agent
 
+# DPDK/SPDK needs pyelftools now
+# Longer term, we should upload our RPM build of this to the public packaging
+# site and install it by RPM, but for now, we'll just pip install it
+USER daos_server:daos_server
+RUN pip3 --no-cache-dir --disable-pip-version-check install --user pyelftools
```

I ended up installing python3-pyelftools on my Fedora box.  I wonder if there is a similar package so we could avoid pip altogether

---

Fedora and Ubuntu seem to have packages, CentOS doesn't for 7 or 8.  @tanabarr landed this change last week, all this PR does is change it so that it's installed for the user only and cache files are deleted.

---

Right, in most of the build chain we can use our "own rolled" version that we fork and put in daos repo, here (workflows) we are running in github and don't have access to our repo, and the only way to get pyelftools in python3 env is to use Pip. It's a real pain that it's not in epel. SPDK CI doesn't cater for other distros, just el7/8 and they simply install with pip in all the RPM and in-tree builds (of which we have raised a ticket about, because really it should be RPM only for the RPM build workflow).

---

Not something that needs to change, but a personal preference is that I like to do this because it better handles (weird) cases where there are multiple python/pip versions installed
```suggestion
RUN python3 -m pip --no-cache-dir --disable-pip-version-check install --user pyelftools
```